src/v/test_utils/http_imposter.cc

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2022 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#include "test_utils/http_imposter.h"
+
+#include "config/node_config.h"
+#include "vlog.h"
+
+#include <seastar/http/function_handlers.hh>
+
+#include <utility>
+
+static ss::logger http_imposter_log("http_imposter"); // NOLINT
+
+http_imposter_fixture::http_imposter_fixture() {
+    _server = ss::httpd::http_server_control();
+    _server.start().get();
+    ss::ipv4_addr ip_addr = {httpd_host_name.data(), httpd_port_number};
+    _server_addr = ss::socket_address(ip_addr);
+}
+
+http_imposter_fixture::~http_imposter_fixture() { _server.stop().get(); }
```

Comment:
would it be needed to deallocate `_handler` before the server is stopped? What if a request is being handled while this destructor is called?

Reply:
I restructured the code a bit and added a gate to the default content handler, which is waited on before stopping the server. 

The default content handler is shared for all requests so it is not ideal, in the long term we probably want to keep track of all content handlers and wait for them when stopping.

If I understand correctly, deallocation should happen automatically when the server is destroyed as handler is a unique_ptr and a member?

Reply:
on further checking, I have removed this gate, because the seastar http server already wraps each call with a gate which waited on when stopping the server: https://github.com/scylladb/seastar/blob/master/src/http/httpd.cc#L506

Reply:
nice find !